### PR TITLE
Byte widget

### PR DESCRIPTION
**Pull request by mattgibbs**
Here's my implementation of the byte widget.  It is pretty simple, just shows individual bits of a integer value.  There are a few properties to change the way it looks and behaves.  This will satisfy #35.

@laispc I think this is different than your byte widget, because it only allows for two states on each indicator, on or off.  I like the idea for yours, but it makes the widget more complicated to use, and at least a SLAC, most of the requests for this only care about binary states.  @hhslepicka is working on a Symbol widget, that displays a single multi-state value, and I'm hopeful that using several symbol widgets will be a solution that works for your use case.  Stay tuned.

### CODE REVIEW

File: pydm/widgets/byte.py [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+    self._connected = False
+    self._on_color = QColor(0,255,0)
+    self._off_color = QColor(100,100,100)
+    self._disconnected_color = QColor(255,255,255)
+    self._invalid_color = QColor(255,0,255)
+    self._pen_style = Qt.SolidLine
+    self._line_pen = QPen(self._pen_style)
+    self._orientation = Qt.Vertical
+    #This is kind of ridiculous, importing QTabWidget just to get a 4-item enum thats usable in Designer.
+    #PyQt5 lets you define custom enums that you can use in designer with QtCore.Q_ENUMS(), doesnt exist in PyQt4.
```
Comment (hhslepicka) [inline pydm/widgets/byte.py]: One more reason to drop Qt4! 😝

File: designer_plugin.py [hunk 2]
```diff
@@ -17,3 +17,5 @@
 from pydm.widgets.spinbox_qtplugin import PyDMSpinboxplugin
 
 from pydm.widgets.drawing_qtplugin import * 
+
+from pydm.widgets.byte_qtplugin import PyDMByteIndicatorPlugin
```
Comment (hhslepicka) [inline designer_plugin.py]: Maybe just name it PyDMByteWidget?

File: pydm/widgets/byte.py [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+class PyDMBitIndicator(QWidget):
+  def __init__(self, parent=None):
+    super(PyDMBitIndicator, self).__init__(parent)
+    self.setAutoFillBackground(True)
+    self.circle = False
+    self._painter = QPainter()
+    self._brush = QBrush(Qt.SolidPattern)
+    self._pen = QPen(Qt.SolidLine)
+    
+  def paintEvent(self, event):
```
Comment (hhslepicka) [inline pydm/widgets/byte.py]: Would it be possible to take benefit of the PyDMDrawing classes in here or it is better to keep them decoupled?

File: pydm/widgets/byte.py [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+  Whether the most significant bit is at the start or end of the widget.
+  """)
+  def bigEndian(self):
+    return self._big_endian
+  
+  @bigEndian.setter
+  def bigEndian(self, is_big_endian):
+    self._big_endian = is_big_endian
+    if self._big_endian:
+      self.layout().setOriginCorner(Qt.BottomLeftCorner)
```
Comment (hhslepicka) [inline pydm/widgets/byte.py]: Very smart use of the OriginCorner to deal with the bigEndian 👍

File: pydm/widgets/byte.py (hunk 3 above)
Comment (mattgibbs) [inline pydm/widgets/byte.py]: I could go either way, but in this case, the indicator is so simple that keeping it separate isn't really duplicating too much code.  The drawing widgets do so much more stuff that I don't think the indicator needs, and I think overriding the drawing class behavior to do the simple color changes I need would be more complicated than just doing it from scratch.

**Pull request merged**